rust-crates/symblib-capi/src/rangeextr.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{FfiResult, StatusCode, SymblibSlice, SymblibString};
+use std::ffi::{c_char, c_void, CStr, OsString};
+use std::os::unix::ffi::OsStringExt as _;
+use std::path::{Path, PathBuf};
+use symblib::objfile::{self, SymbolSource};
+use symblib::symbconv::RangeExtractor as _;
+use symblib::{dwarf, symbconv as sc, symbfile, VirtAddr};
+
+/// Extract ranges from an executable.
+///
+/// This creates a [`symblib::symbconv::multi`] extractor with all supported
+/// debug symbol formats registered with the following priority:
+///
+/// 1) DWARF
+/// 2) Go symbols
+/// 3) ELF debug symbols
+/// 4) ELF dynamic symbols
+///
+/// This extractor is then run to completion and the visitor is invoked for
+/// every range found in the executable. The user_data pointer is passed to
+/// the visitor untouched and may be NULL.
+#[no_mangle]
+pub unsafe extern "C" fn symblib_rangeextr(
```

Comment:
I know this first iteration is not immediately intended to be used in the profiler itself, however since there are ideas/plans to do this in the future I will just dump my thougths.
This API is not very convenient to use for ontarget symbolization because paths will likely be relative to `/proc/{pid}/root`. Maybe we can add an extra argument `rootfs` so that resolving `debugaltlink` is resolved in the same root fs. Or maybe as an alternative we can create an alternative API accepting file descriptors, so that rust lib does not need to worry about rootfs at all.

Reply:
Thanks, feel free to open an issue so that this doesn't get lost in the noise.